test/functorch/test_eager_transforms.py

```diff
@@ -3380,12 +3381,14 @@ def compute_loss(weights, batch, targets):
                 grad_weights, loss = grad_and_value(compute_loss)(weights, batch, targets)
             else:
                 loss = compute_loss(weights, batch, targets)
-                grad_weights = torch.autograd.grad(loss, weights)
-
-            new_weights = []
-            with torch.no_grad():
-                for grad_weight, weight in zip(grad_weights, weights):
-                    new_weights.append(weight - grad_weight * lr)
+                list_weights = weights if mechanism == "make_functional" else list(weights.values())
+                grad_weights = torch.autograd.grad(loss, list_weights)
+                grad_weights = grad_weights if mechanism == "make_functional" else dict(
+                    zip(weights.keys(), grad_weights))
```

Comment:
nit: pytrees save us from the if then else:
```
flat_weights, spec = tree_flatten(weights)
flat_grad_weights = torch.autograd.grad(loss, list_weights)
grad_weights = tree_unflatten(flat_grad_weights, spec)
```

Reply:
Thanks for spotting this! I updated the tests here and in the previous PR to do this